model-engine/model_engine_server/api/app.py

```diff
@@ -32,12 +33,31 @@
 
 logger = make_logger(logger_name())
 
+# Allows us to make the Uvicorn worker concurrency in model_engine_server/api/worker.py very high
+MAX_CONCURRENCY = 500
+
+concurrency_limiter = MultiprocessingConcurrencyLimiter(
+    concurrency=MAX_CONCURRENCY, fail_on_concurrency_limit=True
+)
+
 
 class CustomMiddleware(BaseHTTPMiddleware):
     async def dispatch(self, request: Request, call_next):
         try:
             LoggerTagManager.set(LoggerTagKey.REQUEST_ID, str(uuid.uuid4()))
-            return await call_next(request)
+            if request.url.path in ["/healthcheck", "/healthz", "/readyz"]:
```

Comment:
nit: would just add a comment here that we intentionally exclude health check routes from the concurrency limiter.